sematic/db/models/resolution.py

```diff
@@ -226,3 +235,23 @@ def validate_new(self):
             raise InvalidResolution(
                 f"New resolution {self.root_id} can't begin in the {self.status} state."
             )
+
+    @property
+    def git_info(self) -> Optional[GitInfo]:
+        if self.git_info_json is None:
+            return None
+
+        json_encodable = json.loads(self.git_info_json)
+        return GitInfo(**json_encodable)
+
+    @git_info.setter
+    def git_info(self, value: Optional[GitInfo]) -> None:
+        if value is None:
+            self.git_info_json = None
+            return
+
+        # git_info_json is not mutable; any update posted to api_client will be rejected
+        # we therefore need to sort the keys
+        # for the same reason, we can't use value_to_json_encodable, because it imposes
+        # the values/types/root_type semantics
+        self.git_info_json = json.dumps(dataclasses.asdict(value), sort_keys=True)
```

Comment:
I think `json.dumps` is not necessary here, a `dict` will do. See `Run.external_jobs`.

Reply:
Please see the code comment above. Please let me know if it needs clarifying.